api/v1alpha1/thanosreceive_types.go

```diff
@@ -103,6 +103,21 @@ type IngesterHashringSpec struct {
 	// StorageSize is the size of the storage to be used by the Thanos Receive StatefulSet.
 	// +kubebuilder:validation:Required
 	StorageSize StorageSize `json:"storageSize"`
+	// TenancyConfig is the configuration for the tenancy options.
+	TenancyConfig TenancyConfig `json:"tenancyConfig,omitempty"`
```

Comment:
should this be made optional and a pointer?

Reply:
Sure, but I've moved the tenant matcher type. I don't think we need to explicitly set it, so should be fine